src/SourceBuild/content/test/tests.proj

```diff
@@ -11,7 +11,7 @@
     <!-- Skip scenario tests if the portable OS (determined from the host machine) is different from the target OS
         since the tests require the ability to execute the built SDK. An example of where this would be disabled is
         cross-build of using Mariner to build for Alpine (linux vs linux-musl). -->
-    <_RunScenarioTests Condition="'$(BuildOS)' != 'windows' and '$(__PortableTargetOS.ToLowerInvariant())' != '$(TargetOS.ToLowerInvariant())'">false</_RunScenarioTests>
+    <_RunScenarioTests Condition="'$(BuildOS)' != 'windows' and '$(DotNetBuildSourceOnly)' == 'false' and '$(__PortableTargetOS.ToLowerInvariant())' != '$(TargetOS.ToLowerInvariant())'">false</_RunScenarioTests>
```

Comment:
Can someone help me understand why `__PortableTargetOS` is initialized to `linux-musl` on alpine while `targetOS` is `linux`?   I'm trying to understand if the PortableTargetOS comparison against TargetOS is correct.

Given this is blocking. We should merge and circle back if necessary.

Reply:
Are you asking in the context of SB?

Reply:
> Are you asking in the context of SB?

No, in general.

Reply:
Let's look at an example. In UB cross build scenarios of building Alpine from Mariner, `__PortableTargetOS` is set to `linux` here: https://github.com/dotnet/arcade/blob/be933308b9024d798a9a22c0b8f3c8e3616ffbd8/eng/common/native/init-distro-rid.sh#L111

And `TargetOS` is set to `linux-musl` as a build property here: https://github.com/dotnet/installer/blob/d092a8afb7fe700aafc47fcd7eb2aa6f691de0fa/eng/pipelines/templates/jobs/vmr-build.yml#L327

This build property doesn't impact the value of `targetOS` in the init script that `__PortableTargetOS` is derived from.

/cc @ViktorHofer